test/unit/math/prim/prob/wiener_full_prec_test.cpp

```diff
@@ -0,0 +1,276 @@
+  using stan::math::INFTY;
+  using stan::math::wiener_full_prec_lpdf;
+  double rt = 1, a = 1, v = -1, w = 0.5, t0 = 0.1, sw = 0.2, st0 = 0.1;
+  EXPECT_THROW(wiener_full_prec_lpdf(rt, a, v, w, t0, -1, sw, st0, 1e-4),
+               std::domain_error);
+  EXPECT_THROW(wiener_full_prec_lpdf(rt, a, v, w, t0, INFTY, sw, st0, 1e-4),
+               std::domain_error);
+  EXPECT_THROW(wiener_full_prec_lpdf(rt, a, v, w, t0, -INFTY, sw, st0, 1e-4),
+               std::domain_error);
+  EXPECT_THROW(wiener_full_prec_lpdf(rt, a, v, w, t0, NAN, sw, st0, 1e-4),
+               std::domain_error);
+}
+
+// sw
+TEST(mathPrimScalProbWienerFullPrecScal, invalid_sw) {
+  using stan::math::INFTY;
+  using stan::math::wiener_full_prec_lpdf;
+  double rt = 1, a = 1, v = -1, w = 0.5, t0 = 0.1, sv = 0.2, st0 = 0.1;
+  EXPECT_THROW(wiener_full_prec_lpdf(rt, a, v, w, t0, sv, -1, st0, 1e-4),
+               std::domain_error);
+  EXPECT_THROW(wiener_full_prec_lpdf(rt, a, v, 0.8, t0, sv, 0.5, st0, 1e-4),
+               std::domain_error);  // sw must be smaller than 2*(1-w)
+  EXPECT_THROW(wiener_full_prec_lpdf(rt, a, v, 0.3, t0, sv, 0.7, st0, 1e-4),
+               std::domain_error);  // sw must be smaller than 2*w
+  EXPECT_THROW(wiener_full_prec_lpdf(rt, a, v, w, t0, sv, INFTY, st0, 1e-4),
+               std::domain_error);
+  EXPECT_THROW(wiener_full_prec_lpdf(rt, a, v, w, t0, sv, -INFTY, st0, 1e-4),
+               std::domain_error);
+  EXPECT_THROW(wiener_full_prec_lpdf(rt, a, v, w, t0, sv, NAN, st0, 1e-4),
+               std::domain_error);
+}
+
+// st0
+TEST(mathPrimScalProbWienerFullPrecScal, invalid_st0) {
+  using stan::math::INFTY;
+  using stan::math::wiener_full_prec_lpdf;
+  double rt = 1, a = 1, v = -1, w = 0.5, t0 = 0.1, sv = 0.2, sw = 0.2;
+  EXPECT_THROW(wiener_full_prec_lpdf(rt, a, v, w, t0, sv, sw, -1, 1e-4),
+               std::domain_error);
+  EXPECT_THROW(wiener_full_prec_lpdf(rt, a, v, w, t0, sv, sw, INFTY, 1e-4),
+               std::domain_error);
+  EXPECT_THROW(wiener_full_prec_lpdf(rt, a, v, w, t0, sv, sw, -INFTY, 1e-4),
+               std::domain_error);
+  EXPECT_THROW(wiener_full_prec_lpdf(rt, a, v, w, t0, sv, sw, NAN, 1e-4),
+               std::domain_error);
+}
+
+// CHECK THAT ALL VALID SCALAR TYPES ARE ACCEPTED
+template <typename F>
+void check_scalar_types_2(F& f, double value, double res, double deriv) {
+  // - f: Function with a single parameter exposed, all others
+  // have to be scalars
+  // - value: value to be used for the parameter
+  // - res: expected result of calling `f` with `value`
+  // - deriv: expected result of partial of f with respect to the
+  // parameter in `value`
+  using stan::math::var;
+  double err_tol = 2e-6;
+
+  // type double
+  EXPECT_NEAR(f(value), res, err_tol);
+
+  // type var with derivative
+  var value_var = value;
+  var result_var = f(value_var);
+  result_var.grad();
+  EXPECT_NEAR(value_of(result_var), res, err_tol);
+  EXPECT_NEAR(value_var.adj(), deriv, err_tol);
+}
```

Comment:
Any tests/functions involving `var` types should be in the `rev` test folder

Reply:
Done.